sympy/physics/control/lti.py

```diff
@@ -0,0 +1,76 @@
+from sympy import (
+    Basic, Matrix, latex, ShapeError, Mul,
+    ImmutableMatrix, MutableMatrix, MutableDenseMatrix
+)
+from sympy.polys.polytools import degree
+
+__all__ = ['TransferFunction']
+
+_matrixTypes = (
+    Matrix, ImmutableMatrix, MutableMatrix, MutableDenseMatrix)
+
+
+class TransferFunction(Basic):
+    def __new__(cls, args):
+        # constructor is from a given transfer function.
+        if  isinstance(args, (Matrix, ImmutableMatrix, MutableMatrix)):
+            G = args
+            obj = Basic.__new__(cls, G)
+            obj.G = args
+            return obj
+        else:
+            raise TypeError("Provided argument is of unsupported type.")
+
+    def series(self, other):
+        if not isinstance(other, TransferFunction):
+            raise TypeError("Argument must be of type TransferFunction, not {}.".
+                format(type(other)))
+        # assert matching shapes.
+        if not self.G.shape[0] == other.G.shape[1]:
+            raise ShapeError("Dimensions of the input of the argument and the output of"
+                "the system must match.")
+
+        return TransferFunction(self.G * other.G)
+
+    def parallel(self, other):
+        if not isinstance(other, TransferFunction):
+            raise TypeError("Argument must be of type TransferFunction, not {}.".
+                format(type(other)))
+        # assert matching shapes.
+        if not ((self.G.shape[1] == other.G.shape[1]) and
+                (self.G.shape[0] == other.G.shape[0])):
+            raise ShapeError("Dimensions of the input and the output must match.")
+
+        return TransferFunction(self.G + other.G)
+
+    def neg(self):
+        neg_G = self.G
+        for i, j in enumerate(neg_G):
+            num, denom = j.as_numer_denom()
+            neg_G[i] = Mul(-num, 1/denom)
+
+        return TransferFunction(neg_G)
+
+    def solve(self, u):
+        # assert the right shape of u
+        if not u.shape[1] == 1:
+            raise ShapeError("u must be a column vector, not a matrix.")
+        if not self.G.shape[1] == u.shape[0]:
+            raise ShapeError("u must have a length of {}.".format(self.G.shape[1]))
+
+        return self.G * u
+
+    def _repr_latex_(self):
+        return '$' + latex(self.G) + '$'
+
+
+def total_degree(tf):
+    num, denom = tf.as_numer_denom()
+    return degree(num) - degree(denom)
+
+
+def is_proper(m, strict=False):
+    if not strict:
+        return all(total_degree(tf) <= 0 for tf in m)
+    else:
+        return all(total_degree(tf) < 0 for tf in m)
```

Comment:
I don't know if these concepts are well defined for transfer functions in matrix form.
https://en.wikipedia.org/wiki/Proper_transfer_function
But I think that this should have limited scope that should not be applied for any arbitrary sympy expression, so it should better be added as a property

Reply:
I agree. Also, I'm not sure if `total_degree` is the right name for this type of function. Although, Wikipedia defines _relative degree_ of the transfer function as the difference between the degree of the denominator and that of the numerator. Anyway, I'll add `is_proper` as a property.

Reply:
You can make `total_degree` private function and not expose as public API if there are uncertainties.
For making stuff as `@property` I think that no parameter should be needed, so you'd rather define `is_proper` and `is_strictly_proper` separately.

Reply:
`total_degree` is a helper function for `is_proper`, I think we shouldn't add it as a private function.. because it just checks for one rational function not a matrix of rational functions. I'll adjust its logic in `is_proper` and `is_strictly_proper` itself.